src/monsters.c

```diff
@@ -181,16 +188,18 @@ static order *get_money_for_dragon(region * r, unit * u, int wanted)
 
     /* falls der drache launisch ist, oder das regionssilber knapp, greift er alle an */
     n = 0;
-    for (u2 = r->units; u2; u2 = u2->next) {
-        if (inside_building(u2) != u->building && is_guard(u, GUARD_TAX) && u2->faction != u->faction && cansee(u->faction, r, u2, 0)) {
-            int m = get_money(u2);
-            if (m == 0 || is_guard(u2, GUARD_TAX) || attack_chance <= 0.0)
```

Comment:
Wenn du nicht weist was es da macht, dann lösch es doch nicht einfach, sonder verstehe erstmal warum es da ist. Ganz offensichtlich haben die Monster ja auch Bewacher angegriffen.

Wenn du etwas weiter oben im Code schaust, gibt es da für Drachen eine extra Schleife die alle Bewacher in einer Region immer angreift, damit sie Steuern eintreiben könnnen.
Da sie das jetzt auch trotz bewachern könnne, solltest du die Schleife auf zusätzlich "Bewacher und Flieger" erweitern, damit das wieder geht, und am besten Flieger von dem zufälligen Angriff ausnehmen, damit die nicht Doppelt angegriffen werden. Nicht bewachende Flieger haben dann halt Glück und können sich immer vor dem Drachen in Sicherheit bringen.